addons/crm/models/crm_lead.py

```diff
@@ -348,10 +356,20 @@ def _compute_probabilities(self):
                 if was_automated:
                     lead.probability = lead.automated_probability
 
-    @api.depends('planned_revenue', 'probability')
-    def _compute_expected_revenue(self):
+    @api.depends('expected_revenue', 'probability')
+    def _compute_prorated_revenue(self):
         for lead in self:
-            lead.expected_revenue = round((lead.planned_revenue or 0.0) * (lead.probability or 0) / 100.0, 2)
+            lead.prorated_revenue = round((lead.expected_revenue or 0.0) * (lead.probability or 0) / 100.0, 2)
+
+    @api.depends('recurring_revenue', 'recurring_plan')
```

Comment:
Maybe use recurring_plan.number_of_months in triggers? We would like to ensure that changing plan name does not recompute all opportunities; and changing number of months does it.